pkg/collector/process_test.go

```diff
@@ -57,19 +56,19 @@ func TestTCPCollectingProcess_ReceiveTemplateRecord(t *testing.T) {
 		}
 		defer conn.Close()
 		conn.Write(validTemplatePacket)
-		time.Sleep(time.Second)
-		msgChan := cp.GetMsgChan()
-		for range msgChan {
-			messageCount++
-		}
 	}()
+	messageReceived := make(chan bool)
 	go func() {
-		time.Sleep(4 * time.Second)
-		cp.Stop()
+		go cp.Start()
+		for range cp.GetMsgChan() {
+			messageReceived <- true
+		}
 	}()
-	cp.Start()
-	assert.NotNil(t, cp.templatesMap[1], "TCP Collecting Process should receive and store the received template.")
-	assert.Equal(t, 1, messageCount, "Messages should be stored correctly.")
+	time.Sleep(4 * time.Second)
```

Comment:
do we need this? Could we wait on messageReceived channel and close the collector process?